sdk/core/src/main/java/com/klaviyo/core/config/KlaviyoConfig.kt

```diff
@@ -169,11 +169,14 @@ object KlaviyoConfig : Config {
             }
 
             val permissions = applicationContext!!.packageManager.getPackageInfoCompat(
-                applicationContext!!.packageName, PackageManager.GET_PERMISSIONS
+                applicationContext!!.packageName,
+                PackageManager.GET_PERMISSIONS
             ).requestedPermissions ?: emptyArray()
 
-            if (Manifest.permission.ACCESS_NETWORK_STATE !in permissions) {
-                throw MissingPermission(Manifest.permission.ACCESS_NETWORK_STATE)
+            requiredPermissions.forEach { permission ->
```

Comment:
semi-related, we require two different network permissions, so this checks both.

Reply:
Is caught by anything in the sdk or it will crash the app potentially?

Reply:
I think that's ok since we can't do much without internet.

Reply:
Any of these config errors we throw will crash the app if uncaught, yes. Its either that or the system crashes us later when we try to access those things that are gated by this permission. 
However, we actually declare these permissions in our core sdk manifest. I'm not sure if there is some way to exclude our manifest or something, so I put this in for extra safety

Reply:
Makes sense.